### ISSUE TITLE
.NET (PowerShell) flavor

### ISSUE THREAD

**Issue by zett42** [firasdib/Regex101#1838]
## Flavor Request

I'm requesting a flavor ".NET (PowerShell)". 

The major difference to ".NET (C#)" is, that you **don't have to escape `"` and `\`**.

The backslash can be used unescaped in **any** string literal. In single-quoted string literals, the double-quotation mark can be used unescaped as well. Likewise, in double-quoted string literals the single-quotation mark can be used unescaped. In mixed cases you can use a _here-string_.

**PowerShell examples:**
~~~powershell
$testInput = '"foo" bar "baz"'
[regex]::Matches( $testInput, '"[^"]+"' ).Value

$testInput = 'foo "baz"'
$testInput -match '\S+\s"baz"'

$testInput = 'foo\baz'
$testInput -match '.*?\\.*'

# Here-string is delimited by @' and '@
$testInput = @'
'foo' "bar"
'@
$testInput -match @'
'.*?' ".*?"
'@
~~~ 

**PowerShell documentation:** [about Quoting Rules](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.2)

**Comment by firasdib** [comment]
It would seem excessive to have an entire flavor if the only difference is the escaping. Perhaps in the code gen..

**Comment by zett42** [comment]
In my opinion, code gen seems to be a good way for beginners to get started only. If all I need is the actual regex, it would seem to be too cumbersome to use. Two clicks to open the code sample for the language I'm interested in, then make out the regex within the code sample and more clicks to highlight and copy it. 

Alternative idea: Could you add more delimiters for .NET, similar to Python?
- `" (C#)`
- `' (PowerShell)`

When `'` is choosen, apply the PowerShell escaping rules.

**Comment by mklement0** [comment]
Unfortunately, it isn't just _escaping_ that is the problem, but also what _options_ are selected by default.

In short: only `gi` should be set by default, to match PowerShell's case-_insensitive_ default behavior, which it overlays on the .NET APIs.

 * As an aside: the `gm` default isn't great for the `.NET (C#)` flavor either, given that `m` is _not_ set by default in .NET regexes; the `g` default makes sense, even though there's no direct .NET equivalent.

Ideally, any other options wouldn't even be _selectable_ in the GUI, because native PowerShell regex features (unlike direct use of the underlying .NET APIs) allow only the use of _inline_ regex options (e.g., `(?s)`). The - nontrivial - alternative would be auto-translate the GUI option selections into their equivalent inline regex options.

Having .NET support in general is great, but for PowerShell users it's currently a non-obvious struggle against the default settings of the `.NET (C#)` flavor: I've provided detailed guidance based on the status quo in [this Stack Overflow answer](https://stackoverflow.com/a/73111570/45375), which illustrates that struggle.

**Comment by ninmonkey** [comment]
> When `'` is choosen, apply the PowerShell escaping rules.

It's more complicated than quoted pairs, for example:

- First there's one double-quoted string starting `“` and ending with `"`

| Codepoint | Rune | Category |
|- | - | -|
| 0x201c |  `“` |    InitialQuotePunctuation |
| 0x22  |  `"` |          OtherPunctuation|

- with an inner `"` to escape the `„`

| Codepoint | Rune | Category |
|- | - | -|
| 0x22  |  `"`  |         OtherPunctuation |
| 0x201e  | `„` |   OpenPunctuation|

![image](https://user-images.githubusercontent.com/3892031/185773006-0b84d523-b658-4919-9803-8da76593cde3.png)

Maybe the UI could give hints or link to docs or examples. Or maybe an opt-in button? I think automatically switching would add complexity,

**Comment by mklement0** [comment]
@ninmonkey, yes, PowerShell is regrettably permissive when it comes to interchangeable use of quoting characters - [this Stack Overflow answer](https://stackoverflow.com/a/55053609/45375) lists all equivalents (including for whitespace and hyphen-like chars.)

Personally, I think it's sufficient for PowerShell support on this site to assume that both the REGULAR EXPRESSION and SUBSTITUTION fields are implicitly `'...'`-enclosed, which the UI should clearly reflect.

Ideally, the need to escape the following `'` equivalents by _doubling_ them would also be recognized, but I don't think the need will arise often in practice.

  *  `‘` - [LEFT SINGLE QUOTATION MARK (`U+2018`)]
  * `’` - [RIGHT SINGLE QUOTATION MARK (`U+2019`)]
  * `‚` - [SINGLE LOW-9 QUOTATION MARK (`U+201A`)]
  * `‛` - [SINGLE HIGH-REVERSED-9 QUOTATION MARK (`U+201B`)]
